src/shared/logger/winstonToolkitLogger.ts

```diff
@@ -101,9 +112,49 @@ export class WinstonToolkitLogger implements Logger, vscode.Disposable {
 
         if (types.isNativeError(message)) {
             coerceNameToString(message)
-            this.logger.log(level, '%O', message, ...meta)
+            this.logger.log(level, '%O', message, ...meta, { logID: this.idCounter })
         } else {
-            this.logger.log(level, message, ...meta)
+            this.logger.log(level, message, ...meta, { logID: this.idCounter })
+        }
+
+        this.logMap[this.idCounter % LOGMAP_SIZE] = {}
+        return this.idCounter++
+    }
+
+    /**
+     * Attempts to get the mapped message corresponding to the provided file and logID. 
+     * Log messages are considered 'stale' after a constant amount of new logs have been added.
+     * 
+     * @param logID  Unique ID associated with every log operation
+     * @param file  Desired output file path. Can use "channel://NAME" and "console://NAME" for non-file transports.
+     * 
+     * @returns  Final log message. Stale or non-existant logs return undefined
+     */
+    public getTrackedLog(logID: number, file: string): string | undefined {
+        // This prevents callers from getting stale logs
```

Comment:
Provide visual feedback when this happens?

Reply:
A notification could be used, though this would be up to the caller

Reply:
Isn't there already a sort of "feedback", in that the log will be opened at line 1?